### PR TITLE
fix(kuma-cp): deprecate 'timeout.grpc' section

### PR DESCRIPTION
**Pull request by lobkovilya**
### Summary

Some timeouts were set only for `http` (like idleTimeout), and some were set only for `grpc` (like maxStreamDuration). Since in Envoy GRPC and HTTP timeouts are configured in the same place, we decided to deprecate `grpc` section and set all timeouts in `http` section.

Also, timeouts on the inbound side were not set explicitly by Kuma and Envoy used its own defaults.

### Full changelog

* configure timeouts explicitly on the inbound side
* change default timeout policy to not use `grpc` section
* add unit tests for `timout_configurer`

### Issues resolved

Fix https://github.com/kumahq/kuma/issues/3630 https://github.com/kumahq/kuma/issues/3629 https://github.com/kumahq/kuma/issues/2520

### Documentation

- [X] Link to the website [documentation PR](https://github.com/kumahq/kuma-website/pull/846)

### Testing

- [X] Unit tests
- [ ] E2E tests
- [ ] Manual testing on Universal
- [ ] Manual testing on Kubernetes

### Backwards compatibility

~- [ ] Update [`UPGRADE.md`](/UPGRADE.md) with any steps users will need to take when upgrading.~
- [ ] Add `backport-to-stable` label if the code follows our [backporting policy](/CONTRIBUTING.md#backporting)

### CODE REVIEW

File: pkg/xds/envoy/listeners/v3/timeout_configurer.go [hunk 1]
```diff
@@ -18,10 +18,6 @@ type TimeoutConfigurer struct {
 }
 
 func (c *TimeoutConfigurer) Configure(filterChain *envoy_listener.FilterChain) error {
-	if c.Conf == nil {
-		return nil
-	}
-
```
Comment (bartsmykla) [inline pkg/xds/envoy/listeners/v3/timeout_configurer.go:24]: Why we are removing this check? Was it unnecessary?

**Review by bartsmykla — APPROVED** [review]
one question, but overall lgtm

File: pkg/xds/envoy/listeners/v3/timeout_configurer.go (hunk 1 above)
Comment (lobkovilya) [reply pkg/xds/envoy/listeners/v3/timeout_configurer.go:24]: Empty conf works well to set all zeros, and I'm using it [here](https://github.com/kumahq/kuma/pull/4365/files#diff-8a0428ca30722cd8c5ae3a9f3842e6cb51b71d2b5f423d4ab838f50b2ea094b0R121)

Comment (bartsmykla) [reply pkg/xds/envoy/listeners/v3/timeout_configurer.go:24]: got it!

**Comment by lahabana** [comment]
@lobkovilya have you seen: https://github.com/kumahq/kuma/issues/2520 ? Seems quite related no?

**Review by lahabana — COMMENTED** [review]
Ok seems sensible to me but I haven't thought through the problem much so will let someone else approve :)

Just checking my understanding:

1. We systematically set inbound timeouts to twice the default timeout policy
2. If someone needs longer timeouts they'll still need to use a TP: this is because it's unsafe not to have timeouts and we don't have an inbound timeout policy
3. There's no difference between H2 and grpc are just the same thing

My remaining open question is what needs to be done to a grpc stream work well (I'm thinking doing something like long-polling)?

**Pull request merged**